tasks.py

```diff
@@ -596,3 +611,78 @@ def _view_messages(queue_type, limit):
             if 'MessageAttributes' in message and 'FailedReason' in message['MessageAttributes']:
                 print("{}: {}".format('ReportedError', message['MessageAttributes']['FailedReason']['StringValue']))
             print()
+
+
+def _schedule_csv_to_json(csv_name, json_name, year=None, semester=None):
+    if year is None:
+        year = str(datetime.now().year)
+    if semester is None:
+        month = datetime.now().month
+        if month < 6:
+            semester = "02"
+        elif month < 9:
+            semester = "03"
+        else:
+            semester = "01"
+    elif semester not in ["01", "02", "03"]:
+        print("Semester must be '01' for fall. '02' for winter, or '03' for summer.")
+        return
+
+    csv_file = open(csv_name, "r")
+    json_file = open(json_name, "w")
+
+    fieldnames = tuple(csv_file.readline().split(',')[:-1])
+
+    reader = csv.DictReader(csv_file, fieldnames)
+
+    data = {}
+
+    for row in reader:
+        del row[None]
```

Comment:
What is this doing? Is the `restkey` arg to DictReader something that would be useful?

Reply:
Updated so that DictReader automatically recognizes fieldnames but this issue remains. The rightmost column in the spreadsheet has some random notes in it and DictReader places this under the fieldname of empty string. Dynamo won't accept this key so I'm actively deleting it. Tried restkey/restval but this don't work as technically the extra notes are in the field with the empty name (maybe those are more for malformed csv?).